client/boot/project/wordpress-com.js

```diff
@@ -247,12 +253,16 @@ export function setupMiddlewares( currentUser, reduxStore ) {
 		if ( isMultiTreeLayout && previousLayoutIsSingleTree ) {
 			debug( 'Re-rendering multi-tree layout' );
 			ReactDom.unmountComponentAtNode( document.getElementById( 'wpcom' ) );
-			renderLayout( context.store );
+			renderLayoutPromise( context.store ).then( () => {
+				next();
+			} );
 		} else if ( ! isMultiTreeLayout && ! previousLayoutIsSingleTree ) {
 			debug( 'Unmounting multi-tree layout' );
 			ReactDom.unmountComponentAtNode( document.getElementById( 'primary' ) );
 			ReactDom.unmountComponentAtNode( document.getElementById( 'secondary' ) );
+			next();
+		} else {
```

Comment:
Originally we had an `if ... else if` statement, and the call to `next()` was outside of the statement, so it would be run unconditionally after the block.

But now, in the `if` block, I need to call `next()` async, after the `Layout` is rendered. So I can't have a synchronous call to `next()` outside of the conditional statement. I needed to handle the `else` case, since we still need to call `next()` if we don't get into the `if` or the `else if` block.